### PR TITLE
Release/1.12.0

### PR DESCRIPTION
**Pull request by v-momoin**
## Description

PR includes commits for " Bump native Android SDK version" and "Update TestApp to use latest SDKs"

### CODE REVIEW

File: TestApp/android/app/build.gradle [hunk 1]
```diff
@@ -126,11 +126,11 @@ android {
 }
 
 dependencies {
-    //TODO change to implementation once RN 0.58 is released
     compile project(':appcenter-push')
     compile project(':appcenter-crashes')
     compile project(':appcenter-analytics')
     compile project(':appcenter')
+    //TODO change to implementation once RN 0.58 is released
```
Comment (guperrot) [inline TestApp/android/app/build.gradle]: the comment should not have been moved

File: TestApp/android/app/src/main/java/com/testapp/MainApplication.java [hunk 2]
```diff
@@ -4,16 +4,16 @@
 import android.util.Log;
```
Comment (guperrot) [inline TestApp/android/app/src/main/java/com/testapp/MainApplication.java]: Revert changes in this file.

File: TestApp/android/app/build.gradle (hunk 1 above)
Comment (v-momoin) [reply TestApp/android/app/build.gradle]: Updated the comment location

**Comment by v-momoin** [comment]
Changelog.md was also added in this PR as a separate Commit. "https://github.com/Microsoft/AppCenter-SDK-React-Native/pull/508/commits/8c2970807d9f96a16ad38ba697d45607eb05143f"

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,22 @@
 # App Center SDK for React Native Change Log
 
+## Version 1.12.0
+
+### AppCenter
+
+* **[Fix]** Fix issue where the SDK source could not be built in a directory that contains escaped characters (applies to all modules).
```
Comment (guperrot) [inline CHANGELOG.md]: Please remove that fix as it applies when building apple sdk from source, not react native SDK.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,22 @@
...
+* **[Fix]** Fix TLS 1.2 configuration for some specific devices running API level <21. The bug did not affect all devices running older API levels, only some models/brands, and prevented any data from being sent.
```
Comment (guperrot) [inline CHANGELOG.md]: Add
```

#### Android

```
before

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,22 @@
...
+
+### AppCenterAnalytics
+
+* **[Feature]** Preparation work for a future change in transmission protocol and endpoint for Analytics data. There is no impact on your current workflow when using App Center.
```
Comment (guperrot) [inline CHANGELOG.md]: remove that feature it was elusive change for macos userId support, our react SDK does not have macos support.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,22 @@
...
+* **[Fix]** Extend the current session instead of starting a new session when sending events from the background. Sessions are also no longer started in background by sending an event or a log from another service such as push, as a consequence the push registration information will be missing from crash events information.
```
Comment (guperrot) [inline CHANGELOG.md]: Add

```

#### Android

```

before

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,22 @@
...
+
+### AppCenterDistribute
+
+* **[Fix]** Fix issue with forcing Chrome to open links when other browsers are the default.
+
+
```
Comment (guperrot) [inline CHANGELOG.md]: remove 1 of the 2 blank lines

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,22 @@
...
+* **[Fix]** Fix issue where the SDK source could not be built in a directory that contains escaped characters (applies to all modules).
+* **[Fix]** Fix TLS 1.2 configuration for some specific devices running API level <21. The bug did not affect all devices running older API levels, only some models/brands, and prevented any data from being sent.
+
+### AppCenterAnalytics
+
+* **[Feature]** Preparation work for a future change in transmission protocol and endpoint for Analytics data. There is no impact on your current workflow when using App Center.
+* **[Fix]** Extend the current session instead of starting a new session when sending events from the background. Sessions are also no longer started in background by sending an event or a log from another service such as push, as a consequence the push registration information will be missing from crash events information.
+
+### AppCenterDistribute
+
```
Comment (guperrot) [inline CHANGELOG.md]: Add

```

#### Android

```
before

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,22 @@
 # App Center SDK for React Native Change Log
 
+## Version 1.12.0
+
+### AppCenter
+
```
Comment (guperrot) [inline CHANGELOG.md]: Add

```
* **[Feature]** AppCenter SDK now supports the User ID string, with a maximum of 256 characters, that applies to crashes and push logs. Settable via `AppCenter.setUserId`.
```

File: CHANGELOG.md (hunk 4 above)
Comment (v-momoin) [reply CHANGELOG.md]: done

File: CHANGELOG.md (hunk 5 above)
Comment (v-momoin) [reply CHANGELOG.md]: done

File: CHANGELOG.md (hunk 6 above)
Comment (v-momoin) [reply CHANGELOG.md]: done

File: CHANGELOG.md (hunk 8 above)
Comment (v-momoin) [reply CHANGELOG.md]: Added

File: CHANGELOG.md (hunk 9 above)
Comment (v-momoin) [reply CHANGELOG.md]: added

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,24 @@
...
+* **[Feature]** AppCenter SDK now supports the User ID string, with a maximum of 256 characters, that applies to crashes and push logs. Settable via `AppCenter.setUserId`.
+
+#### Android
```
Comment (guperrot) [inline CHANGELOG.md]: Always add a blank line after a section

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,24 @@
...
+* **[Fix]** Fix TLS 1.2 configuration for some specific devices running API level <21. The bug did not affect all devices running older API levels, only some models/brands, and prevented any data from being sent.
+
+### AppCenterAnalytics
+
+#### Android
```
Comment (guperrot) [inline CHANGELOG.md]: blank line after section

File: CHANGELOG.md (hunk 7 above)
Comment (guperrot) [reply CHANGELOG.md]: this is not resolved there are 2 blank lines, remove 1 of them.

File: CHANGELOG.md (hunk 8 above)
Comment (guperrot) [reply CHANGELOG.md]: I don't see it.

**Pull request merged**